__tests__/restaurants.test.js

```diff
@@ -0,0 +1,150 @@
+          "name": "Salt & Straw",
+          "website": "https://saltandstraw.com/pages/nw-23",
+        },
+      ]
+    `);
+  });
+
+  it('GET api/v1/restaurants/:id should return the restaurant with nested reviews', async () => {
+    const resp = await request(app).get('/api/v1/restaurants/1');
+    expect(resp.status).toBe(200);
+    expect(resp.body).toMatchInlineSnapshot(`
+      Object {
+        "cost": 1,
+        "cuisine": "American",
+        "id": "1",
+        "image": "https://media-cdn.tripadvisor.com/media/photo-o/05/dd/53/67/an-assortment-of-donuts.jpg",
+        "name": "Pip's Original",
+        "reviews": Array [
+          Object {
+            "detail": "Best restaurant ever!",
+            "id": "1",
+            "restaurant_id": "1",
+            "stars": 5,
+            "user_id": "1",
+          },
+          Object {
+            "detail": "Terrible service :(",
+            "id": "2",
+            "restaurant_id": "1",
+            "stars": 1,
+            "user_id": "2",
+          },
+          Object {
+            "detail": "It was fine.",
+            "id": "3",
+            "restaurant_id": "1",
+            "stars": 4,
+            "user_id": "3",
+          },
+        ],
+        "website": "http://www.PipsOriginal.com",
+      }
+    `);
+  });
+
+  const registerAndLogin = async () => {
+    const agent = request.agent(app);
+    const user = await UserService.create(mockUser);
+    await agent
+      .post('/api/v1/users/sessions')
+      .send({ email: mockUser.email, password: mockUser.password });
+    return [agent, user];
+  };
+
+  it('POST /api/v1/restaurants/:id/reviews should create a new review when logged in', async () => {
+    const [agent] = await registerAndLogin();
+    const resp = await agent.post('/api/v1/restaurants/1/reviews').send({
+      stars: 5,
+      detail: 'This is a new review',
+    });
+    expect(resp.status).toBe(200);
+    expect(resp.body).toMatchInlineSnapshot(`
+      Object {
+        "detail": "This is a new review",
+        "id": "4",
+        "restaurant_id": "1",
+        "stars": 5,
+        "user_id": "4",
+      }
+    `);
+  });
+
+  it('POST /api/v1/restaurants/:id/reviews should return a 401 if not authenticated', async () => {
+    const resp = await request(app).post('/api/v1/restaurants/1/reviews').send({
+      stars: 5,
+      detail: 'This should NOT save',
+    });
+    expect(resp.status).toBe(401);
+  });
+
+  it('DELETE /api/v1/reviews/:id should delete a review', async () => {
+    const [agent] = await registerAndLogin();
+    await agent
+      .post('/api/v1/restaurants/4/reviews')
+      .send({ stars: 1, detail: 'So bad' });
+    const res = await agent
+      .delete('/api/v1/reviews/4')
+      .send({ message: 'deleted successfully' });
+    expect(res.status).toBe(200);
+
+    const deleteCheck = await agent.get('/api/v1/reviews/4');
+    expect(deleteCheck.status).toBe(404);
+  });
```

Comment:
it could be good to check that attempting to delete while not logged in as current user / admin returns not authorized also